### PR TITLE
allow use external authentication config from the rule

### PR DESCRIPTION
**Pull request by AlvaroVega**
issue https://github.com/telefonicaid/perseo-fe/issues/638

- [x] Update doc to explain it
- [x] Obfuscate password from action.parameters.authentication.password when retrieving rule (GET api method)

### CODE REVIEW

**Comment by AlvaroVega** [comment]
time=2022-03-07T12:48:58.351Z | lvl=INFO | corr=20eb5d60-83b0-4206-a6bd-4cd12d7d30d9; node=gZtrs19M_; perseocep=359 | trans=d4fd18d6-df5a-49a7-b86d-66c6ee1e8dde | op=/rules?limit=15&offset=0&options=count | path=/rules | comp=perseo-fe | srv=smartcity | subsrv=/ | from=::ffff:172.17.0.15 | msg=sending response: 200 {"error":null,"data":[{"_id":"621ce60e8fdc5bd29afc7efe","name":"user_temperature_post","description":"New rule description","misc":"","text":"select \"user_temperature_post\" as ruleName, ev.id? as id, ev.temperature__type? as type, ev.temperature? as value, * from pattern [every ev=iotEvent]","VR":"","action":{"type":"post","parameters":{"url":"http://172.17.0.1:10026/v2/entities/${id}/attrs","method":"PATCH","version":2,"authentication":{"host":"myhost","port":"myport","user":"myuser","password":"****"},"headers":{"Content-Type":"application/json","Fiware-Service":"db01","Fiware-ServicePath":"/"},"qs":{"temperature":{"value":"${value}","type":"${type}"}}}},"nosignal":"","subservice":"/","service":"smartcity"},{"_id":"621e0db211fa59b40a6c8c8b","name":"my_no_action_rule","description":"my_no_action_rule","misc":"","text":"expression E {(v1,v2) => max(v1,v2)} select \"my_no_action_rule\" as ruleName, ev.id? as id, ev.temperature__type? as type, ev.temperature? as value, * from pattern [every ev=iotEvent]","VR":"","action":[],"nosignal":"","subservice":"/","service":"smartcity"},{"_id":"6225f8eaed301113a55faf6f","name":"mytimerule","description":"mitimerule","misc":"mitimerule","text":"select \"mytimerule\" as ruleName, *, current_timestamp.toDate() as currentTS from pattern [every timer:interval(10 sec)]","VR":"","action":{"type":"update","parameters":{"id":"timer_example_min_2","type":"ww","attributes":[{"name":"timestamp","type":"Text","value":"${currentTS}"}]}},"nosignal":"","subservice":"/","service":"smartcity"}],"count":3}

File: docs/API/plain_rules.md [hunk 1]
```diff
@@ -387,6 +387,9 @@ the Perseo configuration). The `parameters` map includes the following fields:
     -   [Trusts concept](https://docs.openstack.org/keystone/stein/user/trusts)
```
Comment (fgalan) [inline docs/API/plain_rules.md:387]: CNR entry needed for this PR?

File: docs/API/plain_rules.md [hunk 2]
```diff
@@ -390,6 +390,15 @@ the Perseo configuration). The `parameters` map includes the following fields:
...
     updateAction rule (instead of default authentication defined by configuration) which will be used when a trust token
     should be negotiated.
+    i.e.:  
+    ```
+    authentication = {
+        host: 'ext-keystone',
+        port: '5001',
+        user: 'cep',
+        password: 'ceppassword'
+     }
```
Comment (fgalan) [inline docs/API/plain_rules.md:400]:
```suggestion
    "authentication": {
        "host": "ext-keystone",
        "port": "5001",
        "user": "cep",
        "password: "ceppassword"
     }
```

**Review by fgalan — APPROVED** [review]
LGTM

**Pull request merged**